pkg/bgpv1/agent/nodespecer.go

```diff
@@ -21,11 +23,17 @@ type nodeSpecer interface {
 	PodCIDRs() ([]string, error)
 	Labels() (map[string]string, error)
 	Annotations() (map[string]string, error)
+	Run(ctx context.Context)
 }
 
 type kubernetesNodeSpecer struct {
-	Name       string
-	NodeLister v1.NodeLister
+	Name         string
+	NodeLister   v1.NodeLister
+	nodeInformer cache.SharedIndexInformer
```

Comment:
Why exactly these changes? Is this just code refactoring?

Reply:
Yes. We are moving the starting of the informer from the BGP CP constructor to the nodespecer run function. Since nodespecer is consuming the informer, it seems logical that it should own this reference.

Reply:
Okay i think that makes sense.